sinch/core/ports/http_transport.py

```diff
@@ -18,6 +20,21 @@ def request(self, endpoint: HTTPEndpoint) -> HTTPResponse:
         pass
 
     def authenticate(self, endpoint, request_data):
+        if endpoint.HTTP_AUTHENTICATION in (HTTPAuthentication.BASIC.value, HTTPAuthentication.OAUTH.value):
```

Comment:
Authentication "names" are just keywords from OAS files
We can imagine having an API `XXX` using same `Basic` keyword but based onto but requiring credentials different from project[key, id]
This kind of validation should be performed at domain level to ensure the domain is validating it own requirements

Reply:
TBH I'm considering both options - validation  at domain level vs validation at transport level validation. Both solutions have pros and cons.
For now I would like to stick with this solution. I'm going to decide what's best after I finish `service_plan_id` version of the SMS API. Probably refactor will be needed anyway.

Reply:
up to you but IMO: from an architecture and separation of concern point of view having `core` tied to domains implementation details will be a technical debt 
This is tied to domain implementation details because of `core` is now hardcoded to manage authentication requirements according to very specific configuration details but only the domain [must] know which and how to c
onsume them.